faction.cpp

```diff
@@ -69,6 +69,23 @@ const string qs[] = {
 };
 const string *QuitStrs = qs;
 
+void to_json(json &j, const FactionEvent &e) {
+	j = json{{"message", e.message}, {"category", e.category}};
+	if (e.unit != nullptr) {
+		j["unit"] = e.unit->build_json_descriptor();
+	}
+	if (e.region != nullptr) {
+		j["region"] = e.region->basic_region_data();
+	}
```

Comment:
This is the piece of code which pretty much necessitated adding the level field to the region.